### PR TITLE
randomness #6: dkg manager update from randomnet

### PR DESCRIPTION
**Pull request by zjma**

### CODE REVIEW

File: dkg/src/dkg_manager/mod.rs [hunk 1]
```diff
@@ -117,60 +113,128 @@ impl<DKG: DKGTrait> DKGManager<DKG> {
...
-                    self.process_dkg_txn_pulled_notification(dkg_txn).await
+                    self.process_dkg_txn_pulled_notification(dkg_txn)
+                        .await
+                        .map_err(|e|anyhow!("[DKG] process_dkg_txn_pulled_notification failed: {e}"))
                 },
                 close_req = close_rx.select_next_some() => {
                     self.process_close_cmd(close_req.ok())
-                }
+                },
+                _ = interval.tick().fuse() => {
```
Comment (danielxiangzl) [inline dkg/src/dkg_manager/mod.rs:184]: Not sure if this is the best practice for logging.. @zekun000 thoughts?

File: dkg/src/dkg_manager/mod.rs [hunk 2]
```diff
@@ -117,60 +113,128 @@ impl<DKG: DKGTrait> DKGManager<DKG> {
...
-        let (agg_trx_tx, mut agg_trx_rx) = aptos_channel::new(QueueStyle::KLAST, 1, None);
-        self.agg_trx_tx = Some(agg_trx_tx);
+            if metadata.dealer_epoch == self.epoch_state.epoch {
+                info!(
+                    epoch = self.epoch_state.epoch,
+                    "Found unfinished and current DKG session. Continuing it."
+                );
+                self.setup_deal_broadcast(start_time_us, &metadata)
+                    .await
+                    .expect("[DKG] setup_deal_broadcast() should be infallible");
```
Comment (danielxiangzl) [inline dkg/src/dkg_manager/mod.rs:147]: Maybe alert instead of expect?

File: dkg/src/transcript_aggregation/mod.rs [hunk 3]
```diff
@@ -27,15 +29,26 @@ impl<S: DKGTrait> Default for TranscriptAggregator<S> {
 }
 
 pub struct TranscriptAggregationState<DKG: DKGTrait> {
+    start_time: Duration,
+    my_addr: AccountAddress,
+    valid_peer_transcript_seen: bool,
```
Comment (danielxiangzl) [inline dkg/src/transcript_aggregation/mod.rs:34]: It is never set to be true?

File: dkg/src/dkg_manager/mod.rs (hunk 2 above)
Comment (zekun000) [reply dkg/src/dkg_manager/mod.rs:147]: the only error is if inner state is not NotStarted?

File: dkg/src/dkg_manager/mod.rs (hunk 1 above)
Comment (zekun000) [reply dkg/src/dkg_manager/mod.rs:184]: it's okay but this will print out the whole transcript bytes?

Comment (zekun000) [reply dkg/src/dkg_manager/mod.rs:184]: ah i see there's a Debug impl for the transcript

File: dkg/src/dkg_manager/mod.rs [hunk 4]
```diff
@@ -117,60 +113,128 @@ impl<DKG: DKGTrait> DKGManager<DKG> {
...
+                    .with_label_values(&[self.my_addr.to_hex().as_str(), "epoch_change"])
+                    .observe(secs_since_dkg_start);
+                info!(
+                    epoch = self.epoch_state.epoch,
+                    my_addr = self.my_addr,
+                    secs_since_dkg_start = secs_since_dkg_start,
+                    "[DKG] txn executed and entering new epoch.",
+                );
+
+                drop(vtxn_guard);
```
Comment (zekun000) [inline dkg/src/dkg_manager/mod.rs:236]: no need for explicit drop?

**Review by zekun000 — APPROVED** [review]

**Review by danielxiangzl — APPROVED** [review]

**Pull request merged**